packages/oats-axios-adapter/index.ts

```diff
@@ -42,7 +42,10 @@ export function withAxios(axiosInstance: AxiosInstance): runtime.client.ClientAd
       return assert.fail('cannot decide which server to use from ' + arg.servers.join(', '));
     }
     const server = arg.servers[0];
-    const params = axiosToJson(arg.query);
+    // Create `URLSearchParams` instead of passing raw data to axios.
+    // Otherwise axios will suffix array query parameters with "[]".
+    // See https://github.com/axios/axios/blob/master/lib/helpers/buildURL.js#L42
+    const params = createURLSearchParams(axiosToJson(arg.query));
```

Comment:
now this is all not ideal. I'm open to suggestions. Also `URLSearchParams` is not supported by IE.

Reply:
alternative solution: https://github.com/axios/axios/issues/2840#issuecomment-602992060